src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { IKernel } from '../../../kernels/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { KernelDataViewerDependencyImplementation } from './kernelDataViewerDependencyImplementation';
+import { IDataViewerDependencyService } from './types';
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    private withKernel: IDataViewerDependencyService;
+    constructor(
+        @inject(IApplicationShell) applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) isCodeSpace: boolean
+    ) {
+        this.withKernel = new KernelDataViewerDependencyImplementation(applicationShell, isCodeSpace);
+    }
+
+    async checkAndInstallMissingDependencies(kernel: IKernel): Promise<void> {
+        sendTelemetryEvent(Telemetry.DataViewerUsingKernel);
+        return this.withKernel.checkAndInstallMissingDependencies(kernel);
```

Comment:
This is what I would like to avoid, we shodulh't have to copy this code to send telemetry where ever this method is called, it should be done in the method itself.